visualization/app/codeCharta/util/treeMapHelper.ts

```diff
@@ -4,171 +4,190 @@ import { Vector3 } from "three"
-			return new Vector3(vector.x - mapSize + 0.75 * width, vector.y + height, vector.z - mapSize + length / 2)
-		} else {
-			return new Vector3(vector.x - mapSize + width / 2, vector.y + height, vector.z - mapSize + 0.75 * length)
-		}
+function isVisible(squaredNode: CodeMapNode, isNodeLeaf: boolean, s: State, flattened: boolean): boolean {
+	let isVisible = true
+	if (s.dynamicSettings.focusedNodePath.length > 0) {
+		isVisible = squaredNode.path.includes(s.dynamicSettings.focusedNodePath)
 	}
+	if (squaredNode.isExcluded || (isNodeLeaf && s.appSettings.hideFlatBuildings && flattened)) {
+		isVisible = false
+	}
+	return isVisible
+}
 
-	private static isNodeToBeFlat(squaredNode: HierarchyRectangularNode<CodeMapNode>, s: State): boolean {
-		let flattened = false
-		if (
-			s.appSettings.showOnlyBuildingsWithEdges &&
-			s.fileSettings.edges &&
-			s.fileSettings.edges.filter(edge => edge.visible).length > 0
-		) {
-			flattened = this.nodeHasNoVisibleEdges(squaredNode, s)
-		}
-
-		if (s.dynamicSettings.searchedNodePaths && s.dynamicSettings.searchPattern && s.dynamicSettings.searchPattern.length > 0) {
-			flattened = s.dynamicSettings.searchedNodePaths.size == 0 ? true : this.isNodeNonSearched(squaredNode, s)
-		}
+function getIncomingEdgePoint(width: number, height: number, length: number, vector: Vector3, mapSize: number) {
+	if (width > length) {
+		return new Vector3(vector.x - mapSize + width / 4, vector.y + height, vector.z - mapSize + length / 2)
+	}
+	return new Vector3(vector.x - mapSize + width / 2, vector.y + height, vector.z - mapSize + length / 4)
+}
 
-		flattened = squaredNode.data.isFlattened || flattened
-		return flattened
+function getOutgoingEdgePoint(width: number, height: number, length: number, vector: Vector3, mapSize: number) {
+	if (width > length) {
+		return new Vector3(vector.x - mapSize + 0.75 * width, vector.y + height, vector.z - mapSize + length / 2)
 	}
+	return new Vector3(vector.x - mapSize + width / 2, vector.y + height, vector.z - mapSize + 0.75 * length)
+}
 
-	private static nodeHasNoVisibleEdges(squaredNode: HierarchyRectangularNode<CodeMapNode>, s: State): boolean {
-		return (
-			squaredNode.data.edgeAttributes[s.dynamicSettings.edgeMetric] === undefined ||
-			s.fileSettings.edges.filter(edge => squaredNode.data.path === edge.fromNodeName || squaredNode.data.path === edge.toNodeName)
-				.length == 0
-		)
+function isNodeToBeFlat(codeMapNode: CodeMapNode, s: State): boolean {
+	let flattened = false
+	if (s.appSettings.showOnlyBuildingsWithEdges && s.fileSettings.edges?.filter(edge => edge.visible).length > 0) {
+		flattened = nodeHasNoVisibleEdges(codeMapNode, s)
 	}
 
-	private static isNodeNonSearched(squaredNode: HierarchyRectangularNode<CodeMapNode>, s: State): boolean {
-		return !s.dynamicSettings.searchedNodePaths.has(squaredNode.data.path)
+	if (s.dynamicSettings.searchedNodePaths && s.dynamicSettings.searchPattern && s.dynamicSettings.searchPattern.length > 0) {
```

Comment:
Should this be an `else if` case? `flattened` is always defined inside of this if and as such, if the above if case would also apply, it would be redundant work?

Reply:
Yes, we can return earlier. We just have to make sure, that we first check, if the nodePath doesn't match the current searchTerm. In this case, it's always flattened, no matter what settings we chose.